catalyst/dl/utils/torch.py

```diff
@@ -1,34 +1,43 @@
-from typing import Dict, Tuple, Union  # isort:skip
+from typing import Dict, Tuple, Iterable, Callable  # isort:skip
 import copy
 
 import torch
-from torch import nn, optim
+from torch import nn
 from torch.utils.data.dataloader import default_collate as default_collate_fn
 
 from catalyst.dl import utils
 from catalyst.utils import maybe_recursive_call
-
-_Model = nn.Module
-_Criterion = nn.Module
-_Optimizer = optim.Optimizer
-# noinspection PyProtectedMember
-_Scheduler = optim.lr_scheduler._LRScheduler
+from catalyst.utils.typing import (
+    Criterion, Device, Model, Optimizer, Scheduler
+)
 
 
 def process_components(
-    model: _Model,
-    criterion: _Criterion = None,
-    optimizer: _Optimizer = None,
-    scheduler: _Scheduler = None,
+    model: Model,
+    criterion: Criterion = None,
+    optimizer: Optimizer = None,
+    scheduler: Scheduler = None,
     distributed_params: Dict = None,
-    device: Union[str, torch.device] = None,
-) -> Tuple[_Model, _Criterion, _Optimizer, _Scheduler, torch.device]:
+    device: Device = None,
+) -> Tuple[Model, Criterion, Optimizer, Scheduler, Device]:
+    """
+    Returns the processed model, criterion, optimizer, scheduler and device
+
+    Args:
+        model (Model): torch model
+        criterion (Criterion): criterion function
+        optimizer (Optimizer): optimizer
+        scheduler (Scheduler): scheduler
+        distributed_params (dict, optional): dict with the parameters
+            for distributed and FP16 methond
+        device (Device, optional): device
+    """
     distributed_params = distributed_params or {}
     distributed_params = copy.deepcopy(distributed_params)
     if device is None:
         device = utils.get_device()
 
-    model = maybe_recursive_call(model, "to", device=device)
+    model: Model = maybe_recursive_call(model, "to", device=device)
```

Comment:
btw, I think we can remove `maybe_recursive_call` if we would use nn.ModuleDict as PyTorch model

Reply:
I would like to leave it for this PR, and check in the another